intro-to-R-tidyverse/02-intro_to_ggplot2.Rmd

```diff
@@ -0,0 +1,306 @@
+
+**More ggplot2 resources:**  
++ [handy cheatsheet for ggplot2](https://rstudio.com/wp-content/uploads/2015/03/ggplot2-cheatsheet.pdf)  
++ [ggplot2 courses and books from Tidyverse](https://ggplot2.tidyverse.org/)  
++ [ggplot2 data viz chapter of R for Data Science](https://r4ds.had.co.nz/data-visualisation.html)  
++ [ggplot2 online tutorial](http://r-statistics.co/Complete-Ggplot2-Tutorial-Part1-With-R-Code.html)  
+
+## Set Up
+
+We saved these results to a tab separated values (TSV) file called `gene_results_GSE44971.tsv`.
+It's been saved to the `data` folder. 
+File paths are relative to where this notebook file (.Rmd) is saved.
+So we can reference it later, let's make a variable with our data directory name. 
+
+```{r}
+data_dir <- "data"
+```
+
+Let's declare our output folder name as its own variable. 
+
+```{r}
+plots_dir <- "plots"
+```
+
+We can also create a directory if it doesn't already exist. 
+
+```{r}
+# The if statement here tests whether the plot directory exists and
+# only executes the  expressions between the braces if it does not.
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+```
+
+In this notebook we will be using functions from the Tidyverse set of packages, so we need to load in those functions using `library()`.
+We could load the individual packages we need one at a time, but it is convenient for now to load them all with the `tidyverse` package, which groups many of them together.
+Keep a look out for where we tell you which individual package different functions come from.
+
+```{r}
+library(tidyverse)
+```
+
+## Read in the differential expression analysis results file
+
+Here we are using a `tidyverse` function `read_tsv()` from the `readr` package.
+We are also using `file.path()`, which allows us to specify file locations that will work regardless of the operating system being used. 
+The resulting data.frame will be stored in a variable named `stats_df`.
+
+```{r}
+stats_df <- read_tsv(file.path(
+  data_dir,
+  "gene_results_GSE44971.tsv"
+))
+```
+
+Let's take a look at what this data.frame looks like: 
+
+```{r}
+stats_df
+```
+
+We can take a look at a column individually by using a `$`. 
+Note we are using `head()` so the whole thing doesn't print out. 
+
+```{r}
+head(stats_df$contrast)
+```
+
+If we want to see a specific set of values, we can use brackets with the indices of the values we'd like returned.
+
+```{r}
+stats_df$avg_expression[6:10]
+```
+
+Let's checkout how many rows of data there are per each contrast.
+
+```{r}
+summary(stats_df)
+```
+
+## Set up the dataset
+
+Before we make our plot, we want to calculate a set of new values for each row; transformations of the raw statistics in our table.
+To do this we will use a function from the `dplyr` package called `mutate()` to make a new column of -log10 p values.
+
+```{r}
+stats_df <- mutate(stats_df, # data.frame we'd like to add a variable to
+  neg_log_p_val = -log10(p_value)
```

Comment:
If, as I suspect, the other log is not a log10, we might want to name this something like `neg_log10p` for clarity.